### ISSUE TITLE
PO Localization not working on published site

### ISSUE THREAD

**Issue by pbros** [OrchardCMS/OrchardCore#7839]
Hello,

I'm having trouble with having my Localization PO files being applied on my published site. I have a web project (Company.CMS.Web) and a custom theme (Company.CMS.Theme).

I've created a localization PO file in Company.CMS.Web/Localization/fr/Company.CMS.Theme.po and added the following to my Company.CMS.Web.csproj
```
<ItemGroup>
    <Folder Include="Localization\" />
</ItemGroup>
```
When running on my dev instance using Visual Studio, the localizations work perfectly. However, once deployed to my QA server, they don't.

Anything I am missing?

Also, I tried "finding" the translations in the published DLLs/EXE using dotPeek, but I didn't find the values anywhere. How are the localizations supposed to be packaged?

**Comment by hishamco** [comment]
The should be located in the `Localization` folder, but if you add your own I need to ask @jtkech if there's a way to publish them

**Comment by pbros** [comment]
@hishamco I need to add my own, because I have some custom strings in my theme's views for headers and such.

Would there be another way of configuring localized strings not tied to a specific content item?

**Comment by hishamco** [comment]
> I need to add my own, because I have some custom strings in my theme's views for headers and such.

This is common case

> Would there be another way of configuring localized strings not tied to a specific content item?

Could you please elaborate?

**Comment by pbros** [comment]
> > I need to add my own, because I have some custom strings in my theme's views for headers and such.
> 
> This is common case

@jtkech is there anything that I need to validate to make sure that my custom PO files are included in my publish?
 
> > Would there be another way of configuring localized strings not tied to a specific content item?
> 
> Could you please elaborate?

I am simply looking for alternatives to using PO files in the Orchard Core framework for my headers if I can't get them to work.

**Comment by Skrypt** [comment]
The /Localization folder located in your OrchardCore.Cms.Web project will get erased every time you publish or compile. You need to add a different .po file in one of your themes or module. I think you can also use the /Localization folder but with a different .po file than the one already standing in that folder.

**Comment by hishamco** [comment]
That mean MyModule.Header.po should work, Am I right Jasmin?

**Comment by pbros** [comment]
> I think you can also use the /Localization folder but with a different .po file than the one already standing in that folder.

What do you mean by a different .po file standing in that folder? I currently don't have a .po file or a localization folder in my Theme project, but that might be because I created it with an older version?

**Comment by Skrypt** [comment]
The only module that will add a /Localization folder automatically is the OrchardCore.Cms.Web.
You can also add a /Localization to any other Orchard module or theme.
What I meant is that if you want to add a custom .po file in the OrchardCore.Cms.Web module it will likely be removed when the Localizations are updated on our side. So, to avoid that from happening I suggest that you read this documentation.

https://docs.orchardcore.net/en/dev/docs/reference/modules/Localize/

**Comment by pbros** [comment]
Hi @Skrypt 

Thanks for the info. I was able to get it to work by placing the file as

`CMS.Theme/Localization/fr.po`

The documentation is a a bit unclear, as there are 2 bullets for **For each module and theme all files matching**, but only the first (`[ModuleLocation]/Localization/[CultureName].po`) is correct. I had tried with the other variations, but none had worked
```
/Localization/[ModuleId]/[CultureName].po
/Localization/[ModuleId]-[CultureName].po
/Localization/[CultureName]/[ModuleId].po
```

Thanks again for both your assistance.
